src/main/java/com/comphenix/protocol/injector/collection/OutboundPacketListenerSet.java

```diff
@@ -0,0 +1,88 @@
+package com.comphenix.protocol.injector.collection;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+
+import javax.annotation.Nullable;
+
+import com.comphenix.protocol.PacketType;
+import com.comphenix.protocol.ProtocolLibrary;
+import com.comphenix.protocol.concurrent.PacketTypeListenerSet;
+import com.comphenix.protocol.error.ErrorReporter;
+import com.comphenix.protocol.events.ListenerPriority;
+import com.comphenix.protocol.events.ListeningWhitelist;
+import com.comphenix.protocol.events.PacketContainer;
+import com.comphenix.protocol.events.PacketEvent;
+import com.comphenix.protocol.events.PacketListener;
+
+public class OutboundPacketListenerSet extends PacketListenerSet {
+
+	public OutboundPacketListenerSet(PacketTypeListenerSet mainThreadPacketTypes, ErrorReporter errorReporter) {
+		super(mainThreadPacketTypes, errorReporter);
+	}
+
+	@Override
+	protected ListeningWhitelist getListeningWhitelist(PacketListener packetListener) {
+		return packetListener.getSendingWhitelist();
+	}
+
+	@Override
+	public void invoke(PacketEvent event, @Nullable ListenerPriority priorityFilter) {
+		super.invoke(event, priorityFilter);
+
+		if (event.getPacketType() == PacketType.Play.Server.BUNDLE && !event.isCancelled()) {
+			// unpack the bundle and invoke for each packet in the bundle
+			Iterable<PacketContainer> packets = event.getPacket().getPacketBundles().read(0);
+			List<PacketContainer> outPackets = new ArrayList<>();
+			for (PacketContainer subPacket : packets) {
+				if (subPacket == null) {
+					ProtocolLibrary.getPlugin().getLogger().log(Level.WARNING,
```

Comment:
not a huge deal, but you can use ProtocolLogger.log directly and safe a few chained calls